### ISSUE TITLE
Error: [introspection-engine/connectors/sql-introspection-connector/src/re_introspection.rs:382:14] Could not find relation field User on model Client. 

### ISSUE THREAD

**Issue by tomhoule** [prisma/prisma#12527]
We think the error reporting backend has all the necessary information for reproduction. It may be "state error boop" type issue, but we cannot say for sure.

Command: `prisma db pull`
Version: `3.11.0`
Binary Version: `b371888aaf8f51357c7457d836b86d12da91658b`
Report: https://prisma-errors.netlify.app/report/13712 
OS: `arm64 darwin 20.6.0`

JS Stacktrace:
```
Error: [introspection-engine/connectors/sql-introspection-connector/src/re_introspection.rs:382:14] Could not find relation field User on model Client.
    at ChildProcess.<anonymous> (/.../npm/_npx/2778af9cee32ff87/node_modules/prisma/build/index.js:46144:30)
    at ChildProcess.emit (node:events:520:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
```

Rust Stacktrace:
```
   0: backtrace::backtrace::trace
   1: backtrace::capture::Backtrace::new
   2: user_facing_errors::Error::new_in_panic_hook
   3: user_facing_errors::panic_hook::set_panic_hook::{{closure}}
   4: std::panicking::rust_panic_with_hook
   5: std::panicking::begin_panic_handler::{{closure}}
   6: std::sys_common::backtrace::__rust_end_short_backtrace
   7: _rust_begin_unwind
   8: core::panicking::panic_fmt
   9: core::panicking::panic_display
  10: core::panicking::panic_str
  11: core::option::expect_failed
  12: dml::model::Model::find_relation_field_mut
  13: sql_introspection_connector::re_introspection::enrich
  14: sql_introspection_connector::calculate_datamodel::calculate_datamodel
  15: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  16: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  17: <futures_util::future::either::Either<A,B> as core::future::future::Future>::poll
  18: <futures_util::future::future::Then<Fut1,Fut2,F> as core::future::future::Future>::poll
  19: tokio::runtime::task::harness::poll_future
  20: tokio::runtime::task::raw::poll
  21: std::thread::local::LocalKey<T>::with
  22: tokio::runtime::thread_pool::worker::Context::run_task
  23: tokio::runtime::thread_pool::worker::Context::run
  24: tokio::macros::scoped_tls::ScopedKey<T>::set
  25: tokio::runtime::thread_pool::worker::run
  26: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
  27: tokio::runtime::task::harness::Harness<T,S>::poll
  28: tokio::runtime::blocking::pool::Inner::run
  29: std::sys_common::backtrace::__rust_begin_short_backtrace
  30: core::ops::function::FnOnce::call_once{{vtable.shim}}
  31: std::sys::unix::thread::Thread::new::thread_start
  32: _pthread_jit_write_protect_np

```

**Comment by jkomyno** [comment]
Also happens in 4.1.0, as seen [here](https://prisma-errors.netlify.app/report/14189).

**Comment by pimeys** [comment]
This was fixed with the introspection refactoring in 4.6.0 and 4.7.0.
